applications/asset_tracker/CMakeLists.txt

```diff
@@ -22,9 +22,37 @@ zephyr_library_include_directories(
   src/orientation_detector
   src/ui
   src/cloud_codec
+  src/battery_monitor
   )
 
 # Application sources
 add_subdirectory(src/orientation_detector)
 add_subdirectory(src/ui)
 add_subdirectory(src/cloud_codec)
+add_subdirectory(src/gps_controller)
+if (CONFIG_BATTERY_MONITOR)
+add_subdirectory(src/battery_monitor)
+endif()
+
+include_directories(src/env_sensors)
+
+if (CONFIG_USE_BME680_BSEC)
+	target_sources(app PRIVATE src/env_sensors//bsec.c)
+
+	set (bsec_dir ${CMAKE_CURRENT_SOURCE_DIR}/src/env_sensors/BSEC_1.4.7.3_Generic_Release)
+	include_directories(${bsec_dir}/API)
+	target_sources(app PRIVATE ${bsec_dir}/examples/bsec_integration.c)
+	include_directories(${bsec_dir}/examples)
+	target_sources(app PRIVATE ${bsec_dir}/API/bme680.c)
+
+	set(BSEC_LIB_DIR ${bsec_dir}/algo/bin/Normal_version/gcc/Cortex_M4F)
+	set(BSEC_INCLUDE_DIR ${bsec_dir}/algo/bin/Normal_version/gcc/Cortex_M4F)
+	add_library(bsec_lib STATIC IMPORTED GLOBAL)
+	add_dependencies(bsec_lib math_lib bsec_target)
+	set_target_properties(bsec_lib PROPERTIES IMPORTED_LOCATION "${BSEC_LIB_DIR}/libalgobsec.a")
+	set_target_properties(bsec_lib PROPERTIES INTERFACE_INCLUDE_DIRECTORIES ${BSEC_INCLUDE_DIR})
+	target_link_libraries(bsec_lib INTERFACE -L${LIBC_LIBRARY_DIR})
+	target_link_libraries(app PUBLIC bsec_lib)
+else()
+	target_sources(app PRIVATE src/env_sensors/env_sensors.c)
```

Comment:
I believe we should surround this code with "NORDIC SDK APP" markers.

https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/gs_modifying.html#requirements-for-cmakelists-txt

Perhaps @tejlmand can confirm that the parser can deal with complex situations like this.